testfn.go

```diff
@@ -65,6 +66,10 @@ func (t *testcmd) test(c *cli.Context) error {
 	if err != nil {
 		return err
 	}
+	// get name from directory if it's not defined
+	if ff.Name == "" {
+		ff.Name = filepath.Base(filepath.Dir(fpath)) // todo: should probably make a copy of ff before changing it
+	}
```

Comment:
seems like this `findAndParseFuncfile` and `if ff.Name == ""` is repeated all over, maybe we should look into doing this in front of the commands and handing over this info easier. not necessary here, just pointing this out...